### DISCUSSION TITLE
msfupdate within the Docker version?

### DISCUSSION THREAD

**Discussion by lonniev** [rapid7/metasploit-framework#16323] [Q&A]
Probably just not thinking before asking.

When having run Metasploit as a docker-compose run, how does one then get msfupdate to update the exploits of this running msg instance?

Neither running msfupdate from the docker-compose run command nor from the msf6> prompt work.

What am I forgetting or don't know?

**Comment by jmartin-tech** [comment]
The recommended method to utilize the current docker container is to run `./docker/bin/msfconsole` from a repository checkout.
See the [current README.md](https://github.com/rapid7/metasploit-framework/blob/c63490c0504276c200d1dbe5f8965bbdfddd55b8/docker/README.md) for the Framework container.

I suspect some updates to that README.md are need, here is a method to update the container locally.

To rebuild the container with the code from your current checkout `./docker/bin/msfconsole --rebuild` then start the new container with `./docker/bin/msfconsole`.

`msfupdate` does not have code to support update inside the docker container at this time.

The project is looking into methods to improve the update frequency of the published container since this service is no longer free in dockerhub.

Also `docker-compose build` in a clean repository checkout will generate a local container with the latest code.
